enhancements/cluster-logging/elasticsearch-hot-warm-architecture.md

```diff
@@ -0,0 +1,156 @@
+---
+title: elasticsearch-hot-warm-architecture
+authors:
+  - "@ewolinetz"
+reviewers:
+  - "@jcantrill"
+  - "@bparees"
+  - "@alanconway"
+approvers:
+  - "@jcantrill"
+  - "@bparees"
+  - "@alanconway"
+creation-date: 2020-10-13
+last-updated: 2020-10-13
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# elasticsearch-hot-warm-architecture
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Migration plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of this is to increase the potential retention of our logs and maintain that they can still be queried. Currently our logs are either maintained as a high priority index or they are deleted. We seek to emulate the back-end process that Elasticsearch makes available in the form of [x-pack API](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/index-lifecycle-management-api.html) end points.
+
+## Motivation
+
+Currently our managed Elasticsearch cluster is restricted in the amount of days of retention we can provide in part due to the amount of metadata ES keeps in memory related to its indices. If we are taking advantage of the different index management APIs provided by Elasticsearch we can seek to better tune clusters to increase retention and overall performance.
+
+### Goals
+The specific goals of this proposal are:
+
+* Outline the addition to the API to provide a means to configure WARM policies
+
+* Replace|repurpose the current rollover cronjob to also perform hot to warm transition actions
+
+We will be successful when:
+
+* Users are able to configure WARM policies so that we can optimize their cluster usage
+
+* We are able to transition a user's index from one policy to the next based on their configuration
+
+### Non-Goals
+
+* We will not be marking different nodes for the different policies and reallocating shards based on that
```

Comment:
What is the performance implication of this being a non-goal? Are some nodes going to be handling more load because the lift also warm and cold indices whereas other nodes only hot indices?

Reply:
It means customers don't have extra nodes configured for hosting specific indices. There will end up being the same amount of data in less shards, so I would imagine there would be a slight decrease in metadata usage.